### ISSUE TITLE
Pedantic warnings

### ISSUE THREAD

**Issue by pietercollins** [ariadne-cps/development#184]
We may want to divide warnings into "normal" and "pedantic". The former we should try to eliminate (especially for release versions) and should be the default. The latter are warnings which may be useful in finding bugs, but which we also allow code which violates them. 

Examples of (overly) pedantic warnings are are -Wmissing-declarations and -Woverloaded-virtual . An example of a very useful warning which should always be applied is -Wcast-function-type, which usually indicates a bug.

**Comment by lgeretti** [comment]
In general I prefer to have a fixed set of warnings, rather than enable them to check if anything may be off. This second approach is prone to end up in ignoring the "pedantic" level and just working on the normal level anyway. If we deem a warning not worth being there, let's just remove it once and for all.

**Comment by pietercollins** [comment]
That's fine with me. We can always put them in by hand if we want them for a particular file.

**Comment by lgeretti** [comment]
What do you mean? Shall we remove missing-declarations and overloaded-virtual from the fixed set of warnings then?

**Comment by pietercollins** [comment]
Yes, let's remove them. But we can wait until you've cleaned-up anything that triggers these warnings that you think is useful.
